frontend/components/pricing/PremiumAddonBox.vue

```diff
@@ -9,509 +9,351 @@ import {
-        </template>
+
+    <hr>
+
+    <div class="premium-addon-box__body">
+      <div class="premium-addon-box__price">
+        {{ formatFiatCurrency(isPaymentYearly ? totalYearlyPricePerMonth : totalMonthlyPrice) }}
+      </div>
+      <div class="premium-addon-box__price-description">
+        {{ $t("pricing.per_month") }}
+        {{ isPaymentYearly ? $t("pricing.amount_per_year", { amount: formatFiatCurrency(totalYearlyPrice) }) : '' }} *
       </div>
       <div
-        v-if="isYearly"
-        class="saving-info"
+        v-if="isPaymentYearly"
+        class="premium-addon-box__info-badge"
       >
-        <div>
-          {{ $t("pricing.savings", { amount: prices.saving }) }}
-        </div>
+        <span>
+          {{ $t("pricing.savings", {
+            amount: formatFiatCurrency(yearlySubscriptionSavings, { maximumFractionDigits: 0 }),
+          }) }}
+        </span>
         <BcTooltip
           position="top"
           :fit-content="true"
         >
           <FontAwesomeIcon :icon="faInfoCircle" />
           <template #tooltip>
-            <div class="saving-tooltip-container">
+            <div class="premium-addon-box__info-tooltip">
               {{
                 $t("pricing.savings_tooltip", {
-                  monthly: prices.monthly,
-                  monthly_yearly: prices.monthly_based_on_yearly,
+                  monthly: formatFiatCurrency(totalMonthlyPrice),
+                  monthly_yearly: formatFiatCurrency(totalYearlyPricePerMonth),
                 })
               }}
             </div>
           </template>
         </BcTooltip>
       </div>
-      <div class="quantity-row">
-        <div
-          v-if="addonSubscriptionCount"
-          class="quantity-label"
-        >
-          {{
-            $t("pricing.addons.currently_active", {
-              amount: addonSubscriptionCount,
-            })
-          }}
-        </div>
-        <div
-          v-else
-          class="quantity-setter"
+      <div v-if="addonSubscriptionCount">
+        {{
+          $t("pricing.addons.currently_active", {
+            amount: addonSubscriptionCount,
+          })
+        }}
+      </div>
+      <div
+        v-else
+        class="premium-addon-box__subscription-form"
+      >
+        <fieldset
+          aria-labelledby="subscription-count-row-label"
+          class="premium-addon-box__subscription-form-count-row"
         >
-          <Button
-            class="p-button-icon-only"
-            :disabled="purchaseQuantityButtons.minus.disabled"
-            @click="purchaseQuantityButtons.minus.callback"
+          <BcScreenreaderOnly
+            id="subscription-count-row-label"
+            tag="legend"
+          >
+            {{ $t('pricing.addons.select_quantity') }}
+          </BcScreenreaderOnly>
+          <BcButton
+            class="premium-addon-box__subscription-counter-button"
+            :is-disabled="purchaseQuantity <= 1"
+            @click="purchaseQuantity--"
```

Comment:
issue: confusing minus/plus operator

(this should be autocorrected by eslint)
```suggestion
            @click="purchaseQuantity -= 1"
```

See for example: https://www.crockford.com/code.html (Confusing Pluses)

Reply:
Changed